var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -127,7 +134,9 @@ def install(self, spec, prefix):
                 'complex' if '+complex' in spec else 'real'),
             '--with-shared-libraries=%s' % ('1' if '+shared' in spec else '0'),
             '--with-debugging=%s' % ('1' if '+debug' in spec else '0'),
-            '--with-blas-lapack-dir=%s' % spec['lapack'].prefix
+            '--with-lapack-lib=%s' % spec['lapack'].lapack_static_lib,
```

Comment:
i think we already discussing this in https://github.com/LLNL/spack/pull/1538#discussion_r75065373 . Why creating **two PR's with the same changes** which also do not reflect the title of this PR? Please keep things separately. Since you already modify `petsc` similarly in another PR, do not change the file in this one.